lib/base-compiler.ts

```diff
@@ -305,7 +308,7 @@ export class BaseCompiler implements ICompiler {
     }
 
     getRemote() {
-        if (this.compiler.exe === null && this.compiler.remote) return this.compiler.remote;
+        if (this.compiler.exe === '/dev/null' && this.compiler.remote) return this.compiler.remote;
```

Comment:
If this PR introduces bugs, I wouldn't be surprised if it was stuff like this. As far as I can tell this is the only check of CompilerInfo.exe.

Reply:
I think it might be best here to make `this.compiler.remote` the only condition